### PR TITLE
 Created New Berry Form

### PR DESCRIPTION
**Pull request by fohwaz**

### CODE REVIEW

File: src/main/frontend/client/components/BerriesIndex.js [hunk 1]
```diff
@@ -1,35 +1,40 @@
...
+            const response = await fetch("/api/v1/berries")
+            if (!response.ok) {
+                const errorMessage = `${response.status} (${response.statusText})`
+                const error = new Error(errorMessage)
+                throw error
+            }
+            const responseBody = await response.json()
+            setBerries(responseBody.berries)
+        } catch (err) {
+            console.error(`Error in fetch: ${err.message}`)
```
Comment (bkincart) [inline src/main/frontend/client/components/BerriesIndex.js]: It looks like your formatting settings may not be the same across your group! Please make sure you all have the same so this doesn't keep flip flopping :) please resave with indentation of two spaces, and that should remove this file from your changeset other than the addition of the link

File: src/main/frontend/client/components/NewBerryForm.js [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+    const validForSubmission = () => {
+        let submitErrors = {}
+        const requiredFields = ["name", "imgUrl"]
+        requiredFields.forEach(field => {
+            if (formPayload[field].trim() === "") {
+                submitErrors = {...submitErrors, [field]: "Is Blank"}
+            }
+        })
+        setErrors(submitErrors)
+        return _.isEmpty(submitErrors)
```
Comment (bkincart) [inline src/main/frontend/client/components/NewBerryForm.js]: Great frontend validation!

File: src/main/frontend/client/components/NewBerryForm.js [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+    const handleInputChange = event => {
+        setFormPayload({
+            ...formPayload,
+            [event.currentTarget.name]: event.currentTarget.value
+        })
+    }
+
+    if (shouldRedirect) {
+        return <Redirect push to={`/berries`}/>
+    }
```
Comment (bkincart) [inline src/main/frontend/client/components/NewBerryForm.js]: Add a newline here

File: src/main/java/com/launchacademy/reviews/controllers/BerriesApiV1Controller.java [hunk 4]
```diff
@@ -1,48 +1,60 @@
...
+            throw new BerryNotFoundException();
+        }
+        return new ResponseEntity<>(dataMap, HttpStatus.OK);
+    }
+
+    @PostMapping
+    public ResponseEntity<Map<String, Berry>> addNewBerry(@RequestBody Berry berry) {
+        try {
+            Map<String, Berry> dataMap = new HashMap<>();
+            berryService.save(berry);
```
Comment (bkincart) [inline src/main/java/com/launchacademy/reviews/controllers/BerriesApiV1Controller.java]: Let's make sure we add backend validation here!

**Review by bkincart — CHANGES_REQUESTED** [review]
Two big things:
1. Change your formatting to 2-space indentation and that will remove a lot of the changes from this PR :) 
2. Add backend validation please!

**Review by RockyCifu — APPROVED** [review]
Great job adding in the backend validation and fixing the formatting!

**Pull request merged**